File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        for(int i=0;i<numRows;i++)
        {
            vector<int>v(i+1,1) ;
            // for the 1st row there is one element and since i=0 give i+1 indicating the size
             //by default the number is 1
            for(int j=1;j<i;j++)
            // j is enough to start from 1 as i=0,j=0 is 1 and if i=2 j's val is 0,1,2  
            {
                v[j] = res[i-1][j-1] + res[i-1][j];
            }
            res.push_back(v);
        }
        return res;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        for(int i=0;i<numRows;i++)
        {
            List<Integer> arr = new ArrayList<>();
            for(int j=0;j<=i;j++)
            {
                if(j == 0 || j == i) // first or last element
                {
                    arr.add(1);
                }
                else
                {
                    arr.add(ans.get(i-1).get(j-1) + ans.get(i-1).get(j));
                }
            }
            ans.add(arr);
        }
        return ans;
    }
}